src/clr-angular/forms/checkbox/checkbox-container.ts

```diff
@@ -4,23 +4,120 @@
  */
 
-import { Component } from '@angular/core';
+import { Component, ContentChild, Input, OnDestroy, Optional } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { NgControl } from '@angular/forms';
 
-import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
-import { ControlIdService } from '../common/providers/control-id.service';
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { ClrLabel } from '../common/label';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { NgControlService } from '../common/providers/ng-control.service';
 
 @Component({
   selector: 'clr-checkbox-container',
   template: `
-        <!-- We want the checkbox input to be before the label, always -->
-        <ng-content select="[clrCheckbox]"></ng-content>
-        <ng-content></ng-content>
-        <label *ngIf="_dynamic"></label>
-    `,
-  host: { '[class.checkbox]': 'true' },
-  providers: [ControlIdService],
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [class.clr-control-inline]="clrInline" [ngClass]="controlClass()">
+      <ng-content select="clr-checkbox-wrapper"></ng-content>
+      <div class="clr-subtext-wrapper">
+        <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+        <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle" aria-hidden="true"></clr-icon>
+        <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+      </div>
+    </div>
+  `,
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-form-control-disabled]': 'control?.disabled',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [NgControlService, ControlClassService, IfErrorService],
 })
-export class ClrCheckboxContainer implements DynamicWrapper {
-  // Indicates whether the container is dynamically created by the checkbox input itself
-  _dynamic = false;
+export class ClrCheckboxContainer implements OnDestroy {
+  private subscriptions: Subscription[] = [];
+  invalid = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
+  private inline = false;
+  control: NgControl;
+  // private formGroup: AbstractControl;
+
+  /*
+   * Here we want to support the following cases
+   * clrInline - true by presence
+   * clrInline="true|false" - unless it is explicitly false, strings are considered true
+   * [clrInline]="true|false" - expect a boolean
+   */
+  @Input()
+  set clrInline(value: boolean | string) {
+    if (typeof value === 'string') {
+      this.inline = value === 'false' ? false : true;
+    } else {
+      this.inline = !!value;
+    }
+  }
+  get clrInline() {
+    return this.inline;
+  }
+
+  // @TODO Solve for group validation, which doesn't work now with ngModelGroup
```

Comment:
I'm assuming this is here so it can be used once the blocker is resolved. Correct?

Reply:
This was to keep track of my progress for another day, yes.